### ISSUE TITLE
VB project gives target error, not compiler error

### ISSUE THREAD

**Issue by livarcocc** [dotnet/roslyn#25173]
_From @KathleenDollard on March 1, 2018 21:57_

When building a Visual Basic application via the CLI in 2.1.300-Preview2, compiler errors are not reported to the user. 

## Steps to reproduce

* Create a VB console ap
* Create an error in the application. Mine was (the Program.cs)

```VB
Option Explicit On
Option Infer On

Imports System

Module Program
    Sub Main(args As String())
        Dim a As New A
        Dim b As New B
        a.Foo(42)
        b.Foo(42)
        Dim x = (3,4)
        Dim y = (3,4)
        Console.WriteLine(x = y)
    End Sub
End Module
```

Build via CLI. 

## Expected Result

Errors about my code. This is the result if I target netcoreapp2.0 and use SDK 2.1.3:

```
$ dotnet build
Microsoft (R) Build Engine version 15.5.179.9764 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restoring packages for /Users/kathleen/projects/vbconsoletest/vbconsoletest.vbproj...
  Generating MSBuild file /Users/kathleen/projects/vbconsoletest/obj/vbconsoletest.vbproj.nuget.g.props.
  Generating MSBuild file /Users/kathleen/projects/vbconsoletest/obj/vbconsoletest.vbproj.nuget.g.targets.
  Restore completed in 146.16 ms for /Users/kathleen/projects/vbconsoletest/vbconsoletest.vbproj.
/Users/kathleen/projects/vbconsoletest/Program.vb(15,27): error BC30452: Operator '=' is not defined for types '(Integer, Integer)' and '(Integer, Integer)'. [/Users/kathleen/projects/vbconsoletest/vbconsoletest.vbproj]

Build FAILED.

/Users/kathleen/projects/vbconsoletest/Program.vb(15,27): error BC30452: Operator '=' is not defined for types '(Integer, Integer)' and '(Integer, Integer)'. [/Users/kathleen/projects/vbconsoletest/vbconsoletest.vbproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:02.35
```

## Actual Result

Errors from the target. 

```
$ dotnet build
Microsoft (R) Build Engine version 15.6.82.30579 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 40.03 ms for /Users/kathleen/projects/vbconsoletest/vbconsoletest.vbproj.
/usr/local/share/dotnet/sdk/2.1.300-preview2-008260/Roslyn/Microsoft.VisualBasic.Core.targets(73,5): error MSB6006: "vbc.dll" exited with code 1.[/Users/kathleen/projects/vbconsoletest/vbconsoletest.vbproj]

Build FAILED.

/usr/local/share/dotnet/sdk/2.1.300-preview2-008260/Roslyn/Microsoft.VisualBasic.Core.targets(73,5): error MSB6006: "vbc.dll" exited with code 1.[/Users/kathleen/projects/vbconsoletest/vbconsoletest.vbproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:01.30
```

## info

```
.NET Command Line Tools (2.1.300-preview2-008260)

Product Information:
 Version:            2.1.300-preview2-008260
 Commit SHA-1 hash:  4f2e060ef3

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.13
 OS Platform: Darwin
 RID:         osx.10.13-x64
 Base Path:   /usr/local/share/dotnet/sdk/2.1.300-preview2-008260/

Microsoft .NET Core Shared Framework Host

  Version  : 2.1.0-preview2-26131-06
  Build    : b13a0d5c331f374afd35ded57b9a4b4ab128864c
```

_Copied from original issue: dotnet/sdk#2015_

**Comment by livarcocc** [comment]
I remember a similar issue a while ago for csc. Is this the same problem for vbc?

**Comment by jaredpar** [comment]
Checked on latest  and I'm getting the errors as expected
